met-web/src/components/engagement/old-view/widgets/Video/VideoWidgetView.tsx

```diff
@@ -43,24 +63,77 @@ const VideoWidgetView = ({ widget }: VideoWidgetProps) => {
         }
     };
 
+    const formatVideoDuration = (duration: number) => {
+        let minutes = 0;
+        let hours = 0;
+        let seconds = 0;
+        if (3600 < duration) {
+            hours = Math.floor(duration / 3600);
+            minutes = Math.floor((duration - hours * 3600) / 60);
+            seconds = Math.floor(duration - hours * 3600 - minutes * 60);
+            return `${hours} hr ${minutes} min ${seconds} sec`;
+        } else if (3600 > duration && 60 < duration) {
+            minutes = Math.floor(duration / 60);
+            seconds = Math.floor(duration - minutes * 60);
+            return `${minutes} min ${seconds} sec`;
+        } else if (60 > duration && 0 < duration) {
+            return `${seconds} seconds`;
+        } else {
+            return '';
+        }
+    };
+
+    const getVideoSource = (url: string) => {
+        const hostname = new URL(url).hostname;
+        if ('youtube.com' === hostname || 'youtu.be' === hostname) {
+            return 'YouTube';
+        } else if ('vimeo.com' === hostname) {
+            return 'Vimeo';
+        } else if ('facebook.com' === hostname) {
+            return 'Facebook';
+        } else if ('twitch.tv' === hostname) {
+            return 'Twitch';
+        } else if ('soundcloud.com' === hostname) {
+            return 'SoundCloud';
+        } else if ('mixcloud.com' === hostname) {
+            return 'Mixcloud';
+        } else if ('dailymotion.com' === hostname) {
+            return 'DailyMotion';
+        } else {
+            return 'Unknown';
+        }
+    };
+
+    const getVideoTitle = (player: ReactPlayer, source: string, widgetTitle: string) => {
+        if ('YouTube' === source) {
+            return `${player.getInternalPlayer().videoTitle} (${source} video)`;
+        } else if ('Vimeo' === source) {
+            return `${player.getInternalPlayer().element.title} (${source} video)`;
+        } else if ('Facebook' === source || 'Twitch' === source || 'DailyMotion' === source) {
+            return `${widgetTitle} (${source} video)`; // API doesn't return video title, use widget title.
+        } else if ('SoundCloud' === source || 'Mixcloud' === source) {
```

Comment:
It seems like sound platforms are adding a lot of complexity to your title parsing. Were these added platforms requested by the PO?

Reply:
None of the platforms were added, they were just always enabled. All I'm doing is handling all of the possible inputs. Of course, we can change this by URL checking with YUP on the form if it comes to that.